Zend/zend_compile.c

```diff
@@ -6200,6 +6205,7 @@ void zend_compile_func_decl(znode *result, zend_ast *ast, zend_bool toplevel) /*
 	if (is_method) {
 		zend_bool has_body = stmt_ast != NULL;
 		zend_begin_method_decl(op_array, decl->name, has_body);
+		is___tostring = is___tostring && CG(active_class_entry)->__tostring != NULL;
```

Comment:
This is pretty magic. I'd suggest just doing `zend_bool is_tostring = zend_string_equals_ci(decl->name, "__toString")` above.

Reply:
I did it this way to save doing a case conversion since we would need to casefold the method name before doing the comparison.

Reply:
`_ci()` does a case-insensitive comparison. As this is all ASCII, this is not expensive enough to be concerned about.

Reply:
got it, updated